File: app/src/main/java/com/zaclimon/acetv/ui/settings/ForceFitScreenGuidedFragment.java
package com.zaclimon.acetv.ui.settings;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v17.leanback.app.GuidedStepFragment;
import android.support.v17.leanback.widget.GuidanceStylist;
import android.support.v17.leanback.widget.GuidedAction;

import com.zaclimon.acetv.R;
import com.zaclimon.acetv.util.Constants;

import java.util.List;

/**
 * Created by isaac on 17-07-31.
 */

public class ForceFitScreenGuidedFragment extends GuidedStepFragment {

    @Override
    public GuidanceStylist.Guidance onCreateGuidance(Bundle savedInstanceState) {

        SharedPreferences sharedPreferences = getActivity().getSharedPreferences(Constants.ACE_TV_PREFERENCES, Context.MODE_PRIVATE);

        String title = getString(R.string.force_video_fit_title);
        String description = getString(R.string.force_video_fit_description);
        String breadcrumb;
        boolean forceFitActivated = sharedPreferences.getBoolean(Constants.VIDEO_FIT_SCREEN_PREFERENCE, false);

        if (forceFitActivated) {
            breadcrumb = getString(R.string.current_status_text, getString(R.string.activated_text));
        } else {
            breadcrumb = getString(R.string.current_status_text, getString(R.string.deactivated_text));
        }

        return (new GuidanceStylist.Guidance(title, description, breadcrumb, null));
    }

    @Override
    public void onCreateActions(List<GuidedAction> actions, Bundle savedInstanceState) {
        GuidedAction.Builder yesAction = new GuidedAction.Builder(getActivity());
        GuidedAction.Builder noAction = new GuidedAction.Builder(getActivity());
        yesAction.title(R.string.yes_text);
        noAction.title(R.string.no_text);
        yesAction.id(GuidedAction.ACTION_ID_YES);
        noAction.id(GuidedAction.ACTION_ID_NO);
        actions.add(yesAction.build());
        actions.add(noAction.build());
    }

    @Override
    public void onGuidedActionClicked(GuidedAction guidedAction) {

        SharedPreferences sharedPreferences = getActivity().getSharedPreferences(Constants.ACE_TV_PREFERENCES, Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        if (guidedAction.getId() == GuidedAction.ACTION_ID_YES) {
            editor.putBoolean(Constants.VIDEO_FIT_SCREEN_PREFERENCE, true);
        } else {
            editor.putBoolean(Constants.VIDEO_FIT_SCREEN_PREFERENCE, false);
        }

        editor.apply();
        getActivity().finish();
    }

}

File: app/src/main/java/com/zaclimon/tsutaeru/ui/settings/ForceFitScreenGuidedFragment.kt
package com.zaclimon.tsutaeru.ui.settings

import android.content.Context
import android.os.Bundle
import android.support.v17.leanback.app.GuidedStepSupportFragment
import android.support.v17.leanback.widget.GuidanceStylist
import android.support.v17.leanback.widget.GuidedAction
import com.zaclimon.tsutaeru.R
import com.zaclimon.tsutaeru.util.Constants

/**
 * [GuidedStepSupportFragment] used to force an audio-visual content to stretch in order to fit the total
 * resolution of a display.
 *
 * @author zaclimon
 */
class ForceFitScreenGuidedFragment : GuidedStepSupportFragment() {

    override fun onCreateGuidance(savedInstanceState: Bundle?): GuidanceStylist.Guidance {
        val sharedPreferences = context?.getSharedPreferences(Constants.TSUTAERU_PREFERENCES, Context.MODE_PRIVATE)
        val title = getString(R.string.force_video_fit_title)
        val description = getString(R.string.force_video_fit_description)
        val isForceFit = sharedPreferences?.getBoolean(Constants.VIDEO_FIT_SCREEN_PREFERENCE, false) ?: false
        val breadcrumb = if (isForceFit) {
            getString(R.string.current_status_text, getString(R.string.activated_text))
        } else {
            getString(R.string.current_status_text, getString(R.string.deactivated_text))
        }

        return GuidanceStylist.Guidance(title, description, breadcrumb, null)
    }

    override fun onCreateActions(actions: MutableList<GuidedAction>, savedInstanceState: Bundle?) {
        val yesAction = GuidedAction.Builder(context).apply {
            title(R.string.yes_text)
            id(GuidedAction.ACTION_ID_YES)
        }.build()

        val noAction = GuidedAction.Builder(context).apply {
            title(R.string.no_text)
            id(GuidedAction.ACTION_ID_NO)
        }.build()

        actions.add(yesAction)
        actions.add(noAction)
    }

    override fun onGuidedActionClicked(action: GuidedAction?) {
        val sharedPreferences = context?.getSharedPreferences(Constants.TSUTAERU_PREFERENCES, Context.MODE_PRIVATE)
        val editor = sharedPreferences?.edit()
        val id = action?.id

        editor?.putBoolean(Constants.VIDEO_FIT_SCREEN_PREFERENCE, (id == GuidedAction.ACTION_ID_YES))
        editor?.apply()
        activity?.finish()
    }
}